llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -731,6 +804,8 @@ class ModuleSplitter {
                  EntryPointsGroupScope Scope)
       : InputModule(std::move(M)), IsSplit(Split) {
     groupEntryPoints(*InputModule, GMap, Scope);
+    if (DeviceGlobals)
+      checkImageScopedDeviceGlobals(*InputModule, GMap);
```

Comment:
I think that the only responsibility of ModuleSplitter should be splitting. So, it is better to add methods to access InputModule and GMap and move this check out of scope of ModuleSplitter.

Reply:
Add accessors for `InputModule` and `GMap` breaks encapsulation, doesn't it? I think the splitter should split, but it can validate is it able to do its work. I don't agree this is a violation of SRP.

Reply:
The `checkImageScopedDeviceGlobals` checks the validity of the module `InputModule` before splitting. The code of the check is implemented as a free function, the Splitter just calls it in its constructor.

Reply:
Do `std::iterator`s make any checks on container values before to provide iterate functionality?

Reply:
I get your point. I think the best solution is to extract the `groupEntryPoints(...)` from the splitter and pass `GMap` directly into the Splitter' constructor.

Reply:
Thank you @mlychkov for your suggestion. I've provided the refactoring in this commit 74caa294ec452995c6f4e100d4f1ae77666c5de4

Reply:
Thank you very much.